ncm-iptables/src/main/perl/iptables.pm

```diff
@@ -830,80 +783,38 @@ sub find_rule {
     return $?;
 }
 
-##########################################################################
-
-##########################################################################
-
-##########################################################################
-sub Configure($$@) {
-##########################################################################
-
+sub Configure {
     my ($self, $config) = @_;
+    local $@;
 
-    my $iptables;
+    $self->debug(5, "Entering method Configure");
 
-
-    ######################################################################
     # Get global components parameters
-    ######################################################################
-    $iptables = $self->GetResource( $path_iptables, $config );
+    my $iptables = $self->GetResource($path_iptables, $config);
     $self->error($@) and return 1 if $?;
 
-    ######################################################################
-    # Create tmpdir if necessary
-    ######################################################################
-    my ($iptc_temp_fh, $iptc_temp);
-    eval {
-        ($iptc_temp_fh, $iptc_temp) = tempfile("ncm-iptables-XXXXX");
-    };
-    $self->error("failed to create temporary iptables file: $@") and return 1 if $@;
-
-    &WriteFile($self, $iptc_temp, $iptables );
-    if ($? > 0 ) {
-	# bad - bail out
-	$self->error($@);
-	return 1;
-    }
-    $self->debug(1,$@);
-
-    my $changes = 0;
-    $changes = LC::Check::file('/etc/sysconfig/iptables',
-			       owner => 'root',
-			       group => 'root',
-			       mode => '0444',
-			       source => "$iptc_temp",
-	);
-    if($changes) {
-	####################################################################
-	# Reload the service - file changed
-	####################################################################
-
-	if ($NoAction) {
-	    $self->info("Would run \"/sbin/service iptables condrestart\"");
-	} else {
-	    # allow no "dangling" file descriptors, this may be executing in a restricted
-	    # targeted SELinux context
-
-	    my $ip_stdouterr;
-	    if (LC::Process::execute([qw(/sbin/service iptables condrestart)],
-				     "stdout" => \$ip_stdouterr,
-				     "stderr" => "stdout") ) {
-		$self->info("ran \"/sbin/service iptables condrestart\"");
-		if($ip_stdouterr) {
-		    $self->info($ip_stdouterr);
-		}
-	    } else {
-		$self->error("command \"/sbin/service iptables condrestart\" failed:\n$ip_stdouterr");
-	    }
-	}
+    my $changes = $self->WriteFile($CONFIG_IPTABLES, $iptables);
+    $self->debug(5, "WriteFile returned $changes");
+    if ($changes) {
+        # Reload the service - file changed
+        if ($NoAction) {
```

Comment:
That's not quite true. In NoAction mode, $changes will be true if the file _would_ be modified. Therefore the service reload does still need to be NoAction aware, and LC::Process is not. Using CAF::Process instead of LC::Process would take care of this automatically.